.github/workflows/publish.yml

```diff
@@ -0,0 +1,93 @@
+---
+name: Publish images to Docker container registries
+
+env:
+    # https://github.com/docker/metadata-action?tab=readme-ov-file#environment-variables
+    DOCKER_METADATA_PR_HEAD_SHA: true
+
+on:
+    workflow_call:
+        inputs:
+            runsOn:
+                description: GitHub Actions Runner image
+                required: true
+                type: string
+            images:
+                description: Images built in build step
+                required: true
+                type: string
+            registry:
+                description: Docker container registry
+                required: true
+                type: string
+
+jobs:
+
+    release:
+        runs-on: ${{ inputs.runsOn }}
+        timeout-minutes: 30
+        strategy:
+            fail-fast: true
+            matrix:
+                target: ["base", "base-with-services", "lab", "full-stack"]
+
+        steps:
+            - uses: actions/checkout@v4
+
+            - name: Login to GitHub Container Registry 🔑
+              uses: docker/login-action@v3
+              with:
+                  registry: ghcr.io
+                  username: ${{ github.actor }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Login to DockerHub 🔑
+              uses: docker/login-action@v3
+              if: inputs.registry == 'docker.io'
+              with:
+                  registry: docker.io
+                  username: ${{ secrets.DOCKER_USERNAME }}
+                  password: ${{ secrets.DOCKER_PASSWORD }}
+
+            - name: Read build variables
+              id: build_vars
+              run: |
+                  vars=$(cat build.json | jq -c '[.variable | to_entries[] | {"key": .key, "value": .value.default}] | from_entries')
+                  echo "vars=$vars" | tee -a "${GITHUB_OUTPUT}"
+
+            - name: Docker meta
+              id: meta
+              uses: docker/metadata-action@v5
+              env: ${{ fromJSON(steps.build_vars.outputs.vars) }}
+              with:
+                  # e.g. ghcr.io/aiidalab/full-stack
+                  images: ${{ inputs.registry }}/${{ github.repository_owner }}/${{ matrix.target }}
+                  tags: |
+                      type=ref,event=pr
+                      type=edge,enable={{is_default_branch}}
+                      type=raw,value={{tag}},enable=${{ github.ref_type == 'tag' && ! startsWith(github.ref_name, 'v') }}
```

Comment:
```suggestion
                      type=raw,value={{tag}},enable=${{ github.ref_type == 'tag' && startsWith(github.ref_name, 'v') }}
```

This was added by accident I guess ;)

Reply:
Nope, this is intentional. This was the original design, allowing to make a special tagged release by using a tag that doesn't start with `v`. This is also mentioned in the README. I am not too convinced that we need it, but I'd remove it in a separate PR, together with the documentation change.

Reply:
I'd suggest to use the tag, I added this in the first place since I want to get both `2024.1001` and `v2024.1001` as tag. Because you mean don't remember whether the `v` need to be used when running `docker` command.

Reply:
Fair enough, I've removed the line.

Reply:
My suggestion was to keep the tag like `v2024.1001`. Since it is a tag that in use such as 
<img width="931" alt="image" src="https://github.com/aiidalab/aiidalab-docker-stack/assets/8831685/c8e96fb6-a960-4bbc-a6a7-54e0e23f9513">

I use it in PSI server deployment. I can make the change but for users which using `v2024.1001` may confused if we remove it.

Reply:
Yes, but that is already handled in the match group below I believe.